### PR TITLE
chore: better docker caching support

### PR DESCRIPTION
**Pull request by pratikbin**
### Before

On change `requirements.txt`, step `COPY req...` will invalidate subsequent layers of cache, and we have to re fetch all the dependencies which required bandwidth and time.

### After

This changes will mount `~/.cache/pip` to save bandwidth and build time.

### CODE REVIEW

**Comment by moyix** [comment]
Hmm, why use the cache dir of the host system root user? Or do I misunderstand what this change does?

**Comment by pratikbin** [comment]
> Hmm, why use the cache dir of the host system root user? Or do I misunderstand what this change does?

You misunderstood, It is mounting cache directory to the paths on every docker build.

File: copilot_proxy/Dockerfile [hunk 1]
```diff
@@ -5,8 +5,8 @@ FROM python:3.8-slim-buster
 WORKDIR /python-docker
 
 COPY requirements.txt requirements.txt
-RUN pip3 install -r requirements.txt
-RUN pip3 install --extra-index-url https://pypi.ngc.nvidia.com regex fire tritonclient[all]
+RUN --mount=type=cache,target=/root/.cache/pip/ pip3 install -r requirements.txt
```
Comment (leemgs) [inline copilot_proxy/Dockerfile:8]: What for?

Comment (pratikbin) [reply copilot_proxy/Dockerfile:8]: Missed appending `regex fire tritonclient[all]` to requirements

**Comment by fdegier** [comment]
Not sure if this is really desired? The best practice is to not have a cache dir for pip in Docker as it increases the image size. Given that requirements don't change often, it also won't invalidate the subsequent layers, right?

**Comment by fdegier** [comment]
Closing this MR as no further explanation is given by the author and the intended improvement / use case is minimal. 

As long as you don't run `docker build --no-cache` and `requirements.txt` hasn't changed, the cache will be re-used.

**Pull request closed without merging**